fission-web-server/library/Fission/Web/Server/IPFS/Pinner/Class.hs

```diff
@@ -0,0 +1,17 @@
+module Fission.Web.Server.IPFS.Pinner.Class (MonadIPFSPinner (..)) where
+
+import           Fission.Prelude
+
+import qualified Fission.Web.Server.IPFS.Cluster.Error as Cluster
+
+import qualified Network.IPFS.Add.Error                as IPFS.Pin
+import qualified Network.IPFS.Types                    as IPFS
+
+type Errors' = OpenUnion
+  '[ IPFS.Pin.Error
+   , Cluster.Error
+   ]
+
+-- | A monad representing the ability to pin a cid to IPFS (cluster or invidual daemon)
+class Monad m => MonadIPFSPinner m where
```

Comment:
Unconvinced of the name of this. I decided I didn't want to abstract out all the way to `MonadIPFS` because then we have to carry that monad around with us anywhere we want a single function. So this Monad is just in charge of pinning some content: whether it's to a cluster or to a peer

Open to rethinking that or just changing the name (just sounds a little clunky, but couldn't think of another term for it)

Reply:
yeah, a bit clunky, but agreed that it's pretty granular right now. We may want to add stuff to this pinner later, so we could go with something like `MonadCIDStore` to handle our CID storage details.

Alternately, if we _want_ the "pin" part to be front-and-center, you can make the `IPFS` part of the name optional:

```haskell
class Monad m => MonadPinner m where
  pin :: IPFS.CID -> m (Either Errors' ())

-- ...

import qualified Fission.Web.Server.IPFS.Pinner.Class as IPFS

foo :: IPFS.MonadPinner m => m ()
```